File: src/HomeWork2/Calculator.java
package HomeWork2;

public class Calculator implements ICalcultor {
    private double firstOperand;
    private double secondOperand;
    private String operatorStr;

    public double getFirst() {
        return firstOperand;
    }

    public double getSecond() {
        return secondOperand;
    }

    public void setFirstOperand(String operandStr) {
        firstOperand = Double.parseDouble(operandStr);
    }

    public void setSecondOperand(String operandStr) {
        secondOperand = Double.parseDouble(operandStr);
    }

    public void setOperator(String operatorStr) {
        this.operatorStr = operatorStr;
    }

    public String getResult() {
        IOperation operation = operation(operatorStr);
        double res = operation.calculate(firstOperand, secondOperand);
        return Double.toString(res);
    }

    private IOperation operation(String operatorStr) {
        switch (operatorStr) {
            case "+":
                return (firstOperand, secondOperand) -> firstOperand + secondOperand;
            case "-":
                return (firstOperand, secondOperand) -> firstOperand - secondOperand;
            case "*":
                return (firstOperand, secondOperand) -> firstOperand * secondOperand;
            case "/":
                return (firstOperand, secondOperand) -> secondOperand != 0 ? firstOperand / secondOperand : 0;
            default:
                return null;
        }
    }
}

File: src/HomeWork7/Calculator.py

class Calculator:
    def __init__(self):
        self.first_operand = None
        self.operator = None
        self.second_operand = None

    def set_first_operand(self, operand):
        self.first_operand = float(operand)

    def set_operator(self, operator):
        self.operator = operator

    def set_second_operand(self, operand):
        self.second_operand = float(operand)

    def get_result(self):
        if self.operator == '+':
            return self.first_operand + self.second_operand
        elif self.operator == '-':
            return self.first_operand - self.second_operand
        elif self.operator == '*':
            return self.first_operand * self.second_operand
        elif self.operator == '/':
            if self.second_operand == 0:
                return "Error: Division by zero"
            return self.first_operand / self.second_operand
        else:
            return "Error: Invalid operator"
